### ISSUE TITLE
Rule "no-unused-variable" does not work anymore without --type-check

### ISSUE THREAD

**Issue by vilic** [palantir/tslint#2450]
### Bug Report

- __TSLint version__: 5.0.0
- __TypeScript version__: 2.2.1
- __Running TSLint via__:  CLI / VSCode

#### TypeScript code being linted

```ts
import * as Net from 'net';

let foo;
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-unused-variable": [
      true
    ]
  }
}
```

#### Actual behavior
No error reported

#### Expected behavior
Report errors

**Comment by vilic** [comment]
Ah,,, I see. It now requires `--type-check`. I didn't turn it on because of another bug.

**Comment by vilic** [comment]
Oh wait, I think a warning is still missing for that if it's not a bug?

**Comment by adidahiya** [comment]
This was one of the breaking changes in [5.0.0](https://github.com/palantir/tslint/releases/tag/5.0.0). What kind of warning are you looking for? Enabled rules that require the type checker but are not invoked with `--type-check` do produce a warning right now.

**Comment by vilic** [comment]
@adidahiya Ah sorry, I just missed that.
